### PR TITLE
[lte][agw] Add cloud configuration for MME congestion control 

### PR DESCRIPTION
**Pull request by ardzoht**
## Summary

- Adding `congestion_control_enabled` to gateway EPC configs on cloud
- Adding `congestion_control_enabled` to MME service config for local config
- Updating `generate_oai_config` to generate mme.conf template config
- Applying clang-tidy suggestions for double const usage on `mme_app_defs.h` and removing dead code 

## Test Plan

- running orc8r locally with local vm to ensure cloud config is streamed down to AGW
- scale testing (500 UEs / 5 attach rate) with `congestion_control_enabled` set to True
```
magma@phy-u6:~$ sudo cat /var/log/mme.log | grep EMM_CAUSE_CONGESTION | wc -l
199
```
- scale testing (500 UEs / 5 attach rate) with `congestion_control_enabled` set to False
```
magma@phy-u6:~$ sudo cat /var/log/mme.log | grep EMM_CAUSE_CONGESTION | wc -l
0
```

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/configs/mme.yml [hunk 1]
```diff
@@ -27,8 +27,10 @@ enable_apn_correction: false
 apn_correction_map_list:
         - imsi_prefix: "00101"
           apn_override: "magma.ipv4"
+
+congestion_control_enabled: true
```
Comment (ulaskozat) [inline lte/gateway/configs/mme.yml]: do we add this here or to gateway.mconfig? I thought the convention is to use mme.yml for configs not streamed down and /etc/magma/gateway.mconfig for the configs streamed down.

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_main.c [hunk 2]
```diff
@@ -59,6 +59,7 @@ static void mme_app_exit(void);
 bool mme_hss_associated = false;
 bool mme_sctp_bounded   = false;
 task_zmq_ctx_t mme_app_task_zmq_ctx;
+bool mme_congestion_control_enabled = false;
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/mme_app/mme_app_main.c]: shall we make this config consistent with the default value in the config?

File: lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme.c [hunk 3]
```diff
@@ -78,8 +78,10 @@ bool s1ap_dump_ue_hash_cb(
 bool hss_associated = false;
 static int indent   = 0;
 task_zmq_ctx_t s1ap_task_zmq_ctx;
-long s1ap_last_msg_latency = 0;
-long s1ap_zmq_th           = LONG_MAX;
+
+bool s1ap_congestion_control_enabled = false;
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme.c]: let's make the default value consistent with the config file.

File: lte/gateway/configs/mme.yml (hunk 1 above)
Comment (ardzoht) [reply lte/gateway/configs/mme.yml]: It is really dependent on the config, I was following the same pattern as we do for `nat_enabled` (give preference to local service config and if not use mconfig value), but thinking about it, local service config not existing will almost never happen. I will update it to only use mconfig

**Review by ulaskozat — COMMENTED** [review]
shall we also initialize at mme_config_init (https://github.com/magma/magma/blob/9ef32a5fee041164cb163ad32217e47590c3ba1e/lte/gateway/c/core/oai/tasks/mme_app/mme_config.c#L238)?

**Comment by rdefosse** [comment]
@ardzoht 

The OAI pipeline run `https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/3424/` is a false positive

A GitHub service was down and as a result, the OAI pipeline was in bypass mode.

Please re-trigger it.
Sorry for the inconvenience
Raphael

**Comment by ardzoht** [comment]
> @ardzoht
> 
> The OAI pipeline run `https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/3424/` is a false positive
> 
> A GitHub service was down and as a result, the OAI pipeline was in bypass mode.
> 
> Please re-trigger it.
> Sorry for the inconvenience
> Raphael

@rdefosse Rebased on master to re-trigger

**Review by ulaskozat — APPROVED** [review]
LGTM

**Review by pshelar — APPROVED** [review]
LGTM

**Review by andreilee — APPROVED** [review]
Orc8r changes look fine

**Pull request merged**